### PR TITLE
HPCC4J-667 DFSClient: Request traceparents should use active span

### PR DESCRIPTION
**Pull request by jpmcmu**
- Changed RowServiceInputStream to use the active span instead of the top level read span for the traceparent on requests

Signed-off-by: James McMullan James.McMullan@lexisnexis.com

## Type of change:
- [X] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).

## Checklist:
- [X] I have created a corresponding JIRA ticket for this submission
- [X] My code follows the code style of this project.
  - [ ] I have applied the Eclipse code-format template provided.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [X] I have read the HPCC Systems CONTRIBUTORS document (https://github.com/hpcc-systems/HPCC-Platform/wiki/Guide-for-contributors).
- [X] The change has been fully tested:
  - [ ] This change does not cause any existing JUnits to fail.
  - [ ] I have include JUnit coverage to test this change
  - [ ] I have performed system test and covered possible regressions and side effects.
- [X] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] This change fixes the problem, not just the symptom

### CODE REVIEW

File: dfsclient/src/main/java/org/hpccsystems/dfs/client/RowServiceInputStream.java [hunk 1]
```diff
@@ -109,7 +109,7 @@ private static StreamContext constructStreamContext(FieldDef rd, FieldDef pRd, i
         return ctx;
     }
 
-    public static final int          DEFAULT_READ_REQUEST_SPAN_BATCH_SIZE = 25;
```
Comment (rpastrana) [inline dfsclient/src/main/java/org/hpccsystems/dfs/client/RowServiceInputStream.java:112]: this should be reverted

File: dfsclient/src/main/java/org/hpccsystems/dfs/client/RowServiceInputStream.java [hunk 2]
```diff
@@ -2395,15 +2393,25 @@ private void makeFetchObject(StringBuilder sb)
...
         final String trace = traceContextHeader != null ? "\"_trace\": { \"traceparent\" : \"" + traceContextHeader + "\" },\n" : "";
         final String versionMsg = RFCCodes.RFCStreamReadCmd + "{ \"command\" : \"version\", \"handle\": \"-1\", " + trace + " \"format\": \"binary\" }";
         return versionMsg;
     }
 
-    private String makeInitialRequest()
+    private String makeInitialRequest(Span span)
     {
+        String traceContextHeader = null;
+        if (span != null) {
```
Comment (rpastrana) [inline dfsclient/src/main/java/org/hpccsystems/dfs/client/RowServiceInputStream.java:2411]: minor formatting issue

File: dfsclient/src/main/java/org/hpccsystems/dfs/client/RowServiceInputStream.java [hunk 3]
```diff
@@ -2395,15 +2393,25 @@ private void makeFetchObject(StringBuilder sb)
         }
     }
 
-    private String makeGetVersionRequest()
+    private String makeGetVersionRequest(Span versionSpan)
     {
+        String traceContextHeader = null;
```
Comment (rpastrana) [inline dfsclient/src/main/java/org/hpccsystems/dfs/client/RowServiceInputStream.java:2398]: I believe getTraceParentHeader will return null if span == null, or invalid. It feels a little cleaner if we change all these call to gettraceparentheader to:

String traceContextHeader = org.hpccsystems.ws.client.utils.Utils.getTraceParentHeader(versionSpan);

File: dfsclient/src/main/java/org/hpccsystems/dfs/client/RowServiceInputStream.java [hunk 4]
```diff
@@ -2529,8 +2547,13 @@ private String makeTokenRequest()
         return sb.toString();
     }
 
-    private String makeCloseHandleRequest()
+    private String makeCloseHandleRequest(Span span)
     {
+        String traceContextHeader = null;
+        if (span != null) {
+            traceContextHeader = org.hpccsystems.ws.client.utils.Utils.getTraceParentHeader(span);
```
Comment (rpastrana) [inline dfsclient/src/main/java/org/hpccsystems/dfs/client/RowServiceInputStream.java:2554]: I believe you noticed an issue where traceparentheader flags were inconsistent , is that still an issue we need to address?

**Review by rpastrana — CHANGES_REQUESTED** [review]
@jpmcmu a couple of comments

**Comment by jpmcmu** [comment]
Look into changing the default sampling flag in getTraceParentHeader() to 01

**Comment by jpmcmu** [comment]
@rpastrana pushed up code review changes

**Review by rpastrana — APPROVED** [review]
@jpmcmu looks fine, we'll tackle the inconsistent sampled flags with this jira: https://hpccsystems.atlassian.net/browse/HPCC4J-668

**Pull request merged**